src/Stateless/StateConfiguration.cs

```diff
@@ -415,7 +415,7 @@ public StateConfiguration OnEntry(Action entryAction, string entryActionDescript
                 Enforce.ArgumentNotNull(entryAction, nameof(entryAction));
                 return OnEntry(
                     t => entryAction(),
-                    entryActionDescription ?? entryAction.TryGetMethodName());
+                    entryActionDescription);
```

Comment:
Could the ultimate private implementation of `OnEntry` accept an `ActionInfo` like this?

```csharp
return OnEntry(t => entryAction(), ActionInfo.Create(entryAction, entryActionDescription));
```